### ISSUE TITLE
embed/place youtube or video over a panorama 

### ISSUE THREAD

**Issue by L00ncan** [mpetroff/pannellum#1137]
i'd love to add a video or youtube or video over .jpg panoramas, i could add a 360p volcano video and it would appear much higher quality only due to the high quality of the .jpg panorama i'm using. I have many .jpg panoramas of volcano hillsides, i'd love to be able to perhaps embed low quality volcanos erupting on my video and place it similar to how you have the "Custom hot spots" can it be done? it would be an absolute game changer for me for sure :/

**Comment by mpetroff** [comment]
> can it be done?

Yes, you should be able to do something like that by expanding on the method used in the custom hot spots example, as you suggested.

**Comment by L00ncan** [comment]
I'm not a coder though:/ I'd sure love to know the code to make https://pannellum.org/documentation/examples/custom-hot-spots/

blend with a youtube background accompanying, willing to use my own 240p video as a hotspot, i have hundreds of beautiful waterfalls i could add video of the waterfall at 240p from a distance mimicking a High quality video with very little size ratio `
<script>
// Create viewer
viewer = pannellum.viewer('panorama', ﻿{
    "type": "equirectangular",
    "panorama": "/images/bma-1.jpg",
    "hotSpots": [
        {
            "pitch": 14.1,
            "yaw": 1.5,
            "cssClass": "custom-hotspot",
            "createTooltipFunc": hotspot,
            "createTooltipArgs": "Baltimore Museum of Art"
        },
        {
            "pitch": -9.4,
            "yaw": 222.6,
            "cssClass": "custom-hotspot",
            "createTooltipFunc": hotspot,
            "createTooltipArgs": "Art Museum Drive"
        },
        {
            "pitch": -0.9,
            "yaw": 144.4,
            "cssClass": "custom-hotspot",
            "createTooltipFunc": hotspot,
            "createTooltipArgs": "North Charles Street"
        }
    ]
});`

Joomla4 upgrades happening slow though for me so i'm still stuck on Joomla3 unable to upgrade my websites :/ Joomla3 core editors are a real pain headache with scripts currently i can only imagine with Joomla4 it only got worse so the smaller my hotspot youtube mp4 Video script the easier it will be later on. I have no idea what the code change would even look like.

**Comment by mpetroff** [comment]
> I'd sure love to know the code to make https://pannellum.org/documentation/examples/custom-hot-spots/

The code for that is all on that page, so I'm not sure what you're asking.

**Comment by L00ncan** [comment]
I don't see any video code, i see code for #css, i'm not a coder, i would think options to add video would be top of the list of requests but! nothing. Even embedding vimeo,youtube have notes under every video. This Panorama player has very little notes how to add video onto the panorama player.
.
I wanting to embed video into the .jpg panorama, no notes anywhere how to do it. the hotspots shrink also as it goes out of view, would be great if i could keep the size of the embed video and remove the shrinking but i'll worry about getting it embedded first

**Comment by mpetroff** [comment]
The example is for custom hot spots, and it includes the complete HTML, JavaScript, and CSS for doing that. Embedding video is something different, so it isn't included in the example for custom hot spots.

If you can't implement the feature you want yourself, you'll have to find / pay someone to do it for you.

**Comment by mpetroff** [comment]
The example is for custom hot spots, and it includes the complete HTML, JavaScript, and CSS for doing that. Embedding video is something different, so it isn't included in the example for custom hot spots.

If you can't implement the feature you want yourself, you'll have to find / pay someone to do it for you.

**Comment by L00ncan** [comment]
The issue is the Panorama expands in size as it rotates and the video hot-spot remains the same size so it appears the video-hot spot is shrinking when it just remains it's same size as the panorama growths in size. Adding video backdrops to these panoramas would be amazing but the growth and shrinkage ruins it. People could add 240p video backgrounds of running water and rain. Surely i'm not the only person who could see how amazing this feature could be!!!
